pkg/util/ec2/ec2.go

```diff
@@ -158,14 +162,53 @@ func getResponse(url string) (*http.Response, error) {
 	if err != nil {
 		return nil, err
 	}
-
-	if res.StatusCode != 200 {
+	_, tokenAlreadySet := headers["X-aws-ec2-metadata-token"]
+	if res.StatusCode == 401 && !tokenAlreadySet {
+		// Retry with a token
+		token, err := getToken()
+		if err != nil {
+			fmt.Println("3")
+			return nil, err
+		}
+		headers["X-aws-ec2-metadata-token"] = token
+		return getResponse(url, method, headers)
+	} else if res.StatusCode != 200 {
 		return nil, fmt.Errorf("status code %d trying to fetch %s", res.StatusCode, url)
 	}
 
 	return res, nil
 }
 
+func getToken() (string, error) {
+	client := http.Client{
+		Timeout: timeout,
+	}
+
+	req, err := http.NewRequest(http.MethodPut, tokenURL, nil)
+	req.Header.Add("X-aws-ec2-metadata-token-ttl-seconds", "60")
```

Comment:
You may want to check err before adding a header.

Reply:
Indeed, will do.